src/startupscriptgenerator/src/common/commandLineFlags.go

```diff
@@ -10,34 +10,52 @@ import (
 	"flag"
 	"fmt"
 	"os"
+	"strings"
 )
 
-func ValidateCommands(versionCommand *flag.FlagSet, scriptCommand *flag.FlagSet, setupEnvCommand *flag.FlagSet) {
+func getCommandNameList(commands []*flag.FlagSet) []string {
+	cmdsList := make([]string, len(commands))
+	for i := 0; i < len(cmdsList); i++ {
+		cmdsList[i] = commands[i].Name()
+	}
+	return cmdsList
+}
+
+func ValidateCommands(commands []*flag.FlagSet) {
 	// Verify that a subcommand has been provided
-	// os.Arg[0] is the main command
-	// os.Arg[1] will be the subcommand
+	// os.Arg[0] is the main command (ex: 'oryx')
+	// os.Arg[1] will be the subcommand (ex: 'create-script')
 	if len(os.Args) < 2 {
+		cmdsList := getCommandNameList(commands)
 		fmt.Println(fmt.Sprintf(
-			"Error: '%s' or '%s' or '%s' subcommand is required",
-			consts.VersionCommandName,
-			consts.SetupEnvCommandName,
-			consts.CreateScriptCommandName))
+			"Error: Invalid execution. Available subcommands: %s",
+			strings.Join(cmdsList, ", ")))
 		os.Exit(1)
 	}
 
 	// Switch on the subcommand
 	// Parse the flags for appropriate FlagSet
 	// FlagSet.Parse() requires a set of arguments to parse as input
 	// os.Args[2:] will be all arguments starting after the subcommand at os.Args[1]
-	switch os.Args[1] {
-	case consts.VersionCommandName:
+	suppliedCommandName := os.Args[1]
+	if suppliedCommandName == consts.VersionCommandName {
 		PrintVersionInfo()
-	case consts.CreateScriptCommandName:
-		scriptCommand.Parse(os.Args[2:])
-	case consts.SetupEnvCommandName:
-		setupEnvCommand.Parse(os.Args[2:])
-	default:
-		flag.PrintDefaults()
-		os.Exit(1)
+	} else {
+		isValidCommand := false
+		for i := 0; i < len(commands); i++ {
+			if suppliedCommandName == commands[i].Name() {
+				isValidCommand = true
+				commands[i].Parse(os.Args[2:])
+				break
+			}
+		}
```

Comment:
This for loop assumes there is only one valid command, do we guarantee such assumption?

Reply:
I did not understand your question. Here we try to find the command which is of the same name as what the end user has provided. If we find it, then we break.

Reply:
Could we avoid for loop and use an existing method to check if the 'commands' array contains 'suppliedCommandName'?

Reply:
`suppliedCommandName` is a string where as the type of the element in `commands` array is not (for example, I am doing `commands[i].Name()`), so I need to go through each element and do a `.Name()` on that command and compare. Are you asking if there is a C# equivalent in Golang for something like this? `commands.FirstOrDefault(cmd => cmd.Name() ==  suppliedCommandName)`...I checked online and seems like it does not have a built-in way to do this.

Reply:
Gotcha, thanks for explaining. The annotations are also helpful for us to understand it better.